server/fleet/software_installer.go

```diff
@@ -175,6 +178,9 @@ type HostSoftwareInstallerResult struct {
 	InstallScriptExitCode *int `json:"-" db:"install_script_exit_code"`
 	// PostInstallScriptExitCode is used internally to determine the output displayed to the user.
 	PostInstallScriptExitCode *int `json:"-" db:"post_install_script_exit_code"`
+	// SelfService indicates that the installation was queued by the
+	// end user and not an administrator
+	SelfService bool `json:"self_service" db:"self_service"`
```

Comment:
I may have missed it but I don't think I've seen where we insert the `self_service` column for the host installation request? Only seen it for if the software installer allows it.

Reply:
Which probably indicates there's a test missing that should cover that (write the host installation request with `self_service = true`, then get it back from the DB and confirm it is `true`).

Reply:
We don't currently have anywhere that sets the column to true, since we don't have the self-service install software api yet

Reply:
This PR is only to update existing APIs to respect the new columns, not to add an API to initiate the self-service installation

Reply:
Gotcha! :+1: